htlcswitch/interfaces.go

```diff
@@ -535,22 +535,46 @@ type AuxTrafficShaper interface {
 // of HTLC payment attempts. It is designed to support both local and remote
 // lifecycle controllers, allowing full control over result storage and cleanup.
 type AttemptStore interface {
+	// Initialize the payment attempt. This should be called before actually
```

Comment:
Nit: Maybe also add: `and guarantees retry safety in case duplicate HTLCs occur before sending them out to the network.`